pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -23,17 +27,37 @@ type executionStartInfo struct {
 	startTime time.Time
 	traceID   uint64
 	spanID    uint64
+	parentID  uint64
+}
+type invocationPayload struct {
+	Headers map[string]string `json:"headers"`
 }
 
 // currentExecutionInfo represents information from the start of the current execution span
 var currentExecutionInfo executionStartInfo
 
 // startExecutionSpan records information from the start of the invocation.
 // It should be called at the start of the invocation.
-func startExecutionSpan(startTime time.Time) {
+func startExecutionSpan(startTime time.Time, rawPayload string) {
 	currentExecutionInfo.startTime = startTime
 	currentExecutionInfo.traceID = random.Uint64()
 	currentExecutionInfo.spanID = random.Uint64()
+	currentExecutionInfo.parentID = 0
+
+	payload := convertRawPayload(rawPayload)
+
+	if payload.Headers != nil {
+		traceID, e1 := convertStrToUnit64(payload.Headers[traceIDHeader])
+		parentID, e2 := convertStrToUnit64(payload.Headers[parentIDHeader])
+
+		//Strconv failures will return 0. Must ensure error is nil before setting IDs
+		if e1 == nil {
```

Comment:
do we need this? since if the conversion fails, it will be set to zero, and if the error is not nil, you're not setting the traceID (and not setting it will default it to zero I think)

Reply:
Max will there ever be a case where we get an invalid payload and don't want to overwrite the traceID value we set at the start of the function?